src/Microsoft.DotNet.CodeFormatter.Analyzers/ExplicitVariableTypeAnalyzer.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.DotNet.CodeFormatter.Analyzers
+{
+    [Export(typeof(DiagnosticAnalyzer))]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    class ExplicitVariableTypeAnalyzer : DiagnosticAnalyzer
```

Comment:
Do analyzers need to be marked `public`? (If not, explicitly mark it `internal`.)